`random.key_impl`wrong return type annotation

### Description

The code

```python
import jax.random as jr

k = jr.key(1)
impl = jr.key_impl(k)
data = jr.key_data(k)

k2 = jr.wrap_key_data(data, impl=impl)
```
when checked with mypy raises the following error
```python
python-3.11.2 ❯ mypy pp.py
pp.py:7: error: Argument "impl" to "wrap_key_data" has incompatible type "Hashable"; expected "str | PRNGSpec | PRNGImpl | None"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

I believe that `key_impl` should return a `PRNGSpec ` instead of an Hashable ?

### System info (python version, jaxlib version, accelerator, etc.)

0.4.31

--- a/jax/_src/random.py
+++ b/jax/_src/random.py
@@ -14,7 +14,7 @@
 
 from __future__ import annotations
 
-from collections.abc import Hashable, Sequence
+from collections.abc import Sequence
 from functools import partial
 import math
 from operator import index
@@ -292,7 +292,7 @@ def _key_impl(keys: KeyArray) -> PRNGImpl:
   keys_dtype = typing.cast(prng.KeyTy, keys.dtype)
   return keys_dtype._impl
 
-def key_impl(keys: KeyArrayLike) -> Hashable:
+def key_impl(keys: KeyArrayLike) -> PRNGSpec:
   typed_keys, _ = _check_prng_key("key_impl", keys, allow_batched=True)
   return PRNGSpec(_key_impl(typed_keys))
 

